src/librbd/io/ImageRequestWQ.cc

```diff
@@ -756,7 +759,12 @@ void *ImageRequestWQ<I>::_void_dequeue() {
       } else {
         // stall IO until the acquire completes
         ++m_io_blockers;
-        m_image_ctx.exclusive_lock->acquire_lock(new C_AcquireLock(this, item));
+        Context *ctx = new C_AcquireLock(this, item);
+        ctx = create_context_callback<
+          Context, &Context::complete,
+          typename std::decay<decltype(*m_image_ctx.exclusive_lock)>::type>(
```

Comment:
What's going on here? Let the compiler deduce the type of `R` automatically (comment applies throughout).

Reply:
Done